taskcluster_util/util/downloader.py

```diff
@@ -54,23 +54,46 @@ def download_latest_artifact(self, task_id, full_filename, dest_dir):
         sys.stdout.flush()
 
         # move file to dest folder
-        abs_dest_dir = os.path.abspath(dest_dir)
-        log.debug('dest dir: {}'.format(abs_dest_dir))
+        abs_dest_dir = os.path.abspath(dest_dir) if dest_dir else os.getcwd()
         if os.path.exists(abs_dest_dir) and (not os.path.isdir(abs_dest_dir)):
-            log.warning('Not a directory: {}'.format(abs_dest_dir))
+            # exist but is not folder
             final_file_path = os.path.abspath(temp_local_file)
-            log.debug('local file: {}'.format(final_file_path))
+            logger.warning('[{}] is not a folder.'.format(abs_dest_dir))
+            logger.debug('local file: [{}]'.format(final_file_path))
+        elif os.path.isdir(abs_dest_dir) and (not os.access(abs_dest_dir, os.W_OK)):
+            # is folder but no write permission
+            final_file_path = os.path.abspath(temp_local_file)
+            logger.warning('Can not download to [{}]. Permission denied.'.format(abs_dest_dir))
+            logger.debug('local file: [{}]'.format(final_file_path))
         else:
+            # not exist or other
             if not os.path.exists(abs_dest_dir):
-                os.makedirs(abs_dest_dir)
-            shutil.copy(temp_local_file, abs_dest_dir)
-            final_file_path = os.path.join(abs_dest_dir, base_filename)
-            log.debug('local file: {}'.format(final_file_path))
+                # not exist
+                try:
+                    os.makedirs(abs_dest_dir)
+                except os.error as e:
+                    logger.debug('errno: [{}], strerror: [{}], filename: [{}]'.format(e.errno, e.strerror, e.filename))
+                    logger.warning('Can not create the folder: [{}]'.format(abs_dest_dir))
+                    logger.warning('The temprory download file: [{}]'.format(temp_local_file))
+                    # do not remove temp folder
+                    return temp_local_file
+            # then try to copy
+            try:
+                logger.info('The destination folder is [{}]'.format(abs_dest_dir))
+                shutil.copy(temp_local_file, abs_dest_dir)
+                final_file_path = os.path.join(abs_dest_dir, base_filename)
+                logger.debug('local file: [{}]'.format(final_file_path))
+            except:
+                logger.debug(e.args)
+                logger.warning('Can not copy to the folder: [{}]'.format(abs_dest_dir))
+                logger.warning('The temprory download file: [{}]'.format(temp_local_file))
+                # do not remove temp folder
+                return temp_local_file
 
         # remove temp folder
         try:
-            shutil.rmtree(self.temp_dir)  # delete directory
+            shutil.rmtree(self.temp_dir)
```

Comment:
Like said in [this comment](https://github.com/askeing/taskcluster-util-python/pull/11/files?diff=split#r35885920), The folder gets deleted when the path exists but it is not folder